libcobblersignatures/__init__.py

```diff
@@ -2,3 +2,221 @@
+        """
+        return self._signaturesjson
+
+    @signaturesjson.setter
+    def signaturesjson(self, value: str):
+        """
+        The setter for the exported data structure.
+
+        :param value: The str which will be loaded by the Python json utility.
+        """
+        if value is None:
+            pass
+        self._signaturesjson = json.loads(value)
+
+    @property
+    def osbreeds(self) -> List[OsBreed]:
+        """
+        This property represents the currently manipulated data structures.
+
+        :return: The list with the content which is currently manipulated.
+        """
+        return self._osbreeds
+
+    @osbreeds.setter
+    def osbreeds(self, value: List[OsBreed]):
+        """
+        The setter for the osbreeds property.
+
+        :param value: The list with the OsBreeds.
+        """
+        self._osbreeds = value
+
+    def importsignatures(self, import_type: ImportTypes, source: str):
+        """
+        This is the main import function.
+
+        :param import_type: This is one of the four options from the ImportTypes Enum
+        :param source: A string containing the data to be parsed into a json in the end
+        """
+        # TODO make sure source is never none and reasonable
+        if import_type == ImportTypes.FILE:
+            self._importsignaturesfile(source)
+        elif import_type == ImportTypes.URL:
+            self._importsignaturesurl(source)
+        elif import_type == ImportTypes.STRING:
+            self.signaturesjson = source
+        else:
+            raise ValueError("Please use on of the four given options for the source!")
+
+    def _importsignaturesfile(self, filepath: str):
+        """
+        Internal function to handle the import of a file based import.
+
+        :param filepath: The relative or absolute path. Additionally this may be all path variations which are accepted
+                         by the Python ``open()`` function.
+        """
+        with open(filepath, 'r') as f:
+            self.signaturesjson = f.read()
+
+    def _importsignaturesurl(self, url: str):
+        """
+        Internal function to handle the import of a URL based import. The data behind it should be a well formed JSON.
+
+        :param url: The URL to load the content from. This parameter is handed to urllib.
+        """
+        response = urllib.request.urlopen(url)
+        data = response.read()
+        self.signaturesjson = data.decode('utf-8')
+
+    def __prepare_export_output(self, sort_keys: bool = False, indent: Union[None, int] = None) -> str:
+        """
+        Convert the internal data to a JSON. Only for internal usage.
+
+        :param sort_keys: If the keys of the dictionary should be sorted to be more human readable.
+        :param indent: If this is something other then ``None`` then the JSON will be pretty printed.
+        """
+        value = {}
+        for breed in self.osbreeds:
+            value[breed.name] = breed.encode()
```

Comment:
Not saying this is better, but it is a valid alternative for short `for`-loops:
```suggestion
       value = {breed.name: breed.encode() for breed in self.osbreeds}
```

Reply:
I think in this case this is desirable because I plan to hide special cases in `encode()`